### DISCUSSION TITLE
new UMAP and mistake label after Reference Mapping

### DISCUSSION THREAD

**Discussion by jintaoxu07** [satijalab/seurat#4040] [Q&A] [answered]
Hello Satija lab,
Thank you for your great work! 
I mapped my data (human blood leukocytes) to the PBMC Reference Dataset in Seurat 4.0. I followed the steps in "Mapping scRNA-seq queries onto reference datasets".  My dataset contains neutrophils, so I computing a new UMAP visualiztion. 
[ref figure.pdf](https://github.com/satijalab/seurat/files/5935653/ref.figure.pdf)
As you can see, most cells except neutrophils are clustered to the reference dataset. However, if I check the predicted cell types, my neutrophils are mistakenly predicted to be CD14+ monocytes. Because of that, lots of other cells are labeled incorrectly. 
My question is: Is there a way to correct this, like manually select cells in the graph and label them correctly? Or should I just remove my neutrophil population from my dataset before map to the reference dataset? 
Thanks!

**Comment by andrewwbutler** [comment, marked as answer]
Hi,

In transferring the labels from the reference to query dataset, we will always assign a cell the label with the highest prediction score. For cell types that are unique to the query dataset, this often is the "next-closest" reference cell type but is not as ideal as knowing their actual identity (e.g. neutrophils). If you know ahead of time which cells are neutrophils, you could remove them before mapping to avoid misannotation. An alternative is to manually relabel though cells after mapping (e.g. using functions like `SetIdent` to map specific cells to new ident labels). This would allow you to keep the umap plot with all the data but enable relabeling of neutrophils specifically.

**Reply by satijalab** [reply]
I'll just echo Andrew's reply and say that in this case I think the method worked as it should. The neutrophils (which are not in the reference) do not cluster with the reference cells, which is exactly how we want the method to perform and demonstrates that Seurat does not over-correct.

However, for cell annotation, we assign the closest cell type (CD14 Monocytes), which have similar transcriptomes to neutrophils. Therefore the mapping procedure is basically annotating neutrophils as a particular subset of CD14 Monocytes , but which do not cluster with reference CD14 Monocytes. This is how we hope and expect the method to perform.
